src/parser/parser.ee.ts

```diff
@@ -98,3 +98,225 @@ export async function parseStream({
     }
   }
 }
+
+/**
+ * Filter files in the zip based on a list of substrings
+ * @param zipFiles List of files in the zip
+ * @param filterList List of substrings to filter the files
+ * @returns The matched file or undefined
+ */
+function filterFiles(
+  zipFiles: { [key: string]: Uint8Array },
+  filterList: string[],
+): Uint8Array | undefined {
+  for (const fileName of Object.keys(zipFiles)) {
+    if (
+      filterList.some((filter) => {
+        return fileName.includes(filter);
+      })
+    ) {
+      return zipFiles[fileName];
+    }
+  }
+  return undefined;
+}
+
+/**
+ * Parse a .docx file and extract text content
+ * @param buffer - The binary content of the .docx file
+ * @returns Extracted text as a string
+ */
+async function parseDocx(buffer: Buffer): Promise<string> {
```

Comment:
I'd propose to put this into it's own file, as the more we add document types the bigger it will be

reading your code will be easier

mirror what you see on the ansearch repo!